src/olympia/abuse/tests/test_models.py

```diff
@@ -1071,7 +1075,47 @@ def test_handle_job_recreated_existing_job(self):
             other_forwarded_job,
             old_job,
         ]
+        assert list(exisiting_escalation_job.abusereport_set.all()) == [report]
         assert report.reload().cinder_job == exisiting_escalation_job
+        assert NeedsHumanReview.objects.filter(
+            is_active=True
+        ).exists()  # it's not cleared
+
+    def test_handle_job_recreated_existing_report_job(self):
+        addon = addon_factory()
+        exisiting_report_job = CinderJob.objects.create(
+            job_id='5678', target_addon=addon
+        )
+        existing_report = AbuseReport.objects.create(
+            guid=addon.guid, cinder_job=exisiting_report_job
+        )
+
+        decision = CinderDecision.objects.create(
+            action=DECISION_ACTIONS.AMO_ESCALATE_ADDON, addon=addon, notes='blah'
+        )
+        old_job = CinderJob.objects.create(
+            job_id='1234', target_addon=addon, decision=decision
+        )
+        report = AbuseReport.objects.create(guid=addon.guid, cinder_job=old_job)
+        NeedsHumanReview.objects.create(
+            version=addon.current_version,
+            reason=NeedsHumanReview.REASONS.ABUSE_ADDON_VIOLATION,
+        )
+
+        old_job.handle_job_recreated(new_job_id='5678')
+
+        old_job.reload()
+        exisiting_report_job.reload()
+        assert old_job.forwarded_to_job == exisiting_report_job
+        assert list(exisiting_report_job.forwarded_from_jobs.all()) == [old_job]
+        assert list(exisiting_report_job.abusereport_set.all()) == [
+            existing_report,
+            report,
+        ]
+        assert report.reload().cinder_job == exisiting_report_job
+        assert not NeedsHumanReview.objects.filter(
+            is_active=True
+        ).exists()  # it's cleared
```

Comment:
Related to the comment I made above: in real life, we'd have a `NeedsHumanReview` with `CINDER_ESCALATION` reason left, right ? We don't here because that's done by the webhook handler but I'd feel better if you added it manually before, and then adjusted that assertion to ensure it's still here and we only cleared the `ABUSE_ADDON_VIOLATION` one.

Reply:
https://github.com/mozilla/addons-server/pull/22683/commits/6657358f084879812247fe4acb07b9ba6958b56a